src/metabase/query_processor/middleware/constraints.clj

```diff
@@ -5,26 +5,27 @@
    [metabase.models.setting :as setting]
    [metabase.util.i18n :refer [deferred-tru]]))
 
-(def ^:private ^:const default-max-results-bare-rows 2000)
-
-;; NOTE: this was changed from a hardcoded var with value of 2000 (now moved to [[default-max-results-bare-rows]])
-;; to a setting in 0.43 the setting, which allows for DB local value, can still be nil, so any places below that used
-;; to reference the former constant value have to expect it could return nil instead
 (setting/defsetting max-results-bare-rows
   (deferred-tru "Maximum number of rows to return specifically on :rows type queries via the API.")
-  :visibility     :authenticated
+  :visibility     :internal
   :type           :integer
-  :database-local :allowed)
+  :default        2000
+  :database-local :allowed
+  :getter         (fn [] (setting/get-value-of-type :integer :max-results-bare-rows)))
 
-(def ^:private max-results
-  "General maximum number of rows to return from an API query."
-  10000)
+(setting/defsetting max-results
+  (deferred-tru "General maximum number of rows to return from an API query.")
+  :visibility     :internal
+  :type           :integer
+  :default        10000
+  :database-local :allowed
+  :getter         (fn [] (setting/get-value-of-type :integer :max-results)))
 
 (defn default-query-constraints
   "Default map of constraints that we apply on dataset queries executed by the api."
   []
-  {:max-results           max-results
-   :max-results-bare-rows (or (max-results-bare-rows) default-max-results-bare-rows)})
+  {:max-results           (setting/get :max-results)
+   :max-results-bare-rows (setting/get :max-results-bare-rows)})
```

Comment:
```suggestion
  {:max-results           (max-results)
   :max-results-bare-rows (max-results-bare-rows)})
```

`defsetting` generates getter functions, and it's preferrable IMO to use them over `setting/get` since it's easier to jump to the definition in the editor